### PR TITLE
testsuite: Test both interpolaton code paths of p3m

### PR DESCRIPTION
**Pull request by fweik**
P3M can optionally use an interpolated window function, this tests both possible code paths.

### CODE REVIEW

File: testsuite/python/coulomb_cloud_wall.py [hunk 1]
```diff
@@ -102,7 +102,15 @@ def compare(self, method_name, energy=True, prefactor=None):
     # Tests for individual methods
 
     if espressomd.has_features(["P3M"]):
-        def test_p3m(self):
+        def test_p3m_direct_caf(self):
```
Comment (KaiSzuttor) [inline testsuite/python/coulomb_cloud_wall.py:108]: please add a docstring

Comment (fweik) [reply testsuite/python/coulomb_cloud_wall.py:108]: They all run the same check with different methods, none of the other test functions have docstrings.

Comment (KaiSzuttor) [reply testsuite/python/coulomb_cloud_wall.py:108]: while I can imagine what happens in ```test_scafacos_p3m``` its hard to grasp the meaning of ```test_p3m_direct_caf```

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
once the tests have a short description it can be merged

**Review by KaiSzuttor — DISMISSED** [review]

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**